ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCompletion/Rules/FSharpRelevanceRule.fs

```diff
@@ -39,6 +41,18 @@ type FSharpRelevanceRule() =
                 if mfv.IsMember && mfv.IsProperty then
                     markRelevance item CLRLookupItemRelevance.FieldsAndProperties
                 else
+                    if fcsLookupItem.FcsSymbolUse.IsFromComputationExpression then
+                        let isCustomOperationPossible =
+                            let reference = context.ReparsedContext.Reference
+                            if isNull reference then false else
+
+                            let refExpr = reference.GetTreeNode().As<IReferenceExpr>()
+                            if isNull refExpr then false else
+
+                            isInsideComputationExpressionForCustomOperation refExpr
```

Comment:
It would be nice to calculate it once if there're many custom operators, so we don't try to traverse the tree for each one again. On the other hand, it looks like the traversal should be short enough, so I'm not sure if this can ever be an actual issue. We could either use something like `lazy` outside the loop, or store the data in the completion context, like `context.GetOrCreateData` (see its other usages in the plugin), if we want to improve it.

Reply:
Done